### ISSUE TITLE
io.py's read_text() blows up when handling an empty directory

### ISSUE THREAD

**Issue by afloresescarcega** [paul-gauthier/aider#89]
Stacktrace:

```
Add the output to the chat? y
Traceback (most recent call last):
  File "/usr/local/bin/aider", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aider/main.py", line 335, in main
    coder.run()
  File "/usr/local/lib/python3.11/site-packages/aider/coders/base_coder.py", line 385, in run
    new_user_message = self.send_new_user_message(new_user_message)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aider/coders/base_coder.py", line 473, in send_new_user_message
    messages += self.get_files_messages()
                ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aider/coders/base_coder.py", line 358, in get_files_messages
    repo_content = self.repo_map.get_repo_map(self.abs_fnames, other_files)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aider/repomap.py", line 107, in get_repo_map
    res = self.choose_files_listing(chat_files, other_files)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aider/repomap.py", line 138, in choose_files_listing
    files_listing = self.get_ranked_tags_map(chat_files, other_files)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aider/repomap.py", line 381, in get_ranked_tags_map
    ranked_tags = self.get_ranked_tags(chat_fnames, other_fnames)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aider/repomap.py", line 313, in get_ranked_tags
    idents = self.get_name_identifiers(fname, uniq=False)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aider/repomap.py", line 248, in get_name_identifiers
    idents = self.get_name_identifiers_uncached(fname)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aider/repomap.py", line 257, in get_name_identifiers_uncached
    content = self.io.read_text(fname)
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aider/io.py", line 142, in read_text
    with open(str(filename), "r", encoding=self.encoding) as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
IsADirectoryError: [Errno 21] Is a directory: '/Users/alexfloresescarcega/REPOS/Minecraft-clone/lib/assimp'```

`~/REPOS/Minecraft-clone/lib/assimp` is an empty directory 

I was able to apply a workaround fix by adding this case
```except IsADirectoryError:
            self.tool_error(f"{filename}: is a directory, not a file")
            return```
            
The workaround still gives an error `/Users/alexfloresescarcega/REPOS/Minecraft-clone/lib/assimp: is a directory, not a file` but allows me to continue with the program.

I would be more than happy to send a fix and write a test if there is a specific fix in mind.

**Comment by paul-gauthier** [comment]
Thanks for trying aider and reporting this issue.

There's a good chance this bug is already resolved as part of issue #57. Can you try installing the latest version (0.8.3) and see if you still have this problem?

**Comment by paul-gauthier** [comment]
I'm going to close this issue for now, but feel free to re-open or file a new issue any time.
